Add ObjectFactory to AttributeContainer (#35068)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

ADD RELEASE FEATURES BELOW
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->

<a name="build-authoring"></a>
### Build authoring improvements

> #### Introduce `AttributeContainer#named`

Reviewer: ```suggestion
#### New `AttributeContainer.named()` method
```

> This release introduces a new API on `AttributeContainer` allowing to create attribute values using the new `named` to not require depending on an ObjectFactory.

Reviewer: Needs javadoc links as well:

```suggestion
This release introduces a new convenience method on `AttributeContainer`, `named()`, which lets you create attribute values directly from the container—there is no need to obtain an `ObjectFactory` first.
```

> Consider the following example demonstrating the new API's behavior:

Reviewer: ```suggestion
This makes attribute assignment more concise while preserving the same semantics as creating a named value via `ObjectFactory`:
```

Author: There are no spaces before the dash, is it expected?

```kotlin
val color = Attribute.of("color", String::class.java)

configurations.resolvable("foo") {
    attributes {

>         attribute(color, named("red"))

Reviewer: ```suggestion
        // before: attribute(color, objects.named(String::class.java, "red"))
        attribute(color, named("red"))
```

ADD RELEASE FEATURES BELOW
vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -->

<a name="build-authoring"></a>
### Build authoring improvements

#### New `AttributeContainer.named()` method

This release introduces a new convenience method on `AttributeContainer`, [`named()`](javadoc/org/gradle/api/attributes/AttributeContainer.html#named(java.lang.Class,java.lang.String)), which can create attribute values directly from the container without requiring a separate `ObjectFactory` instance.

This method makes attribute assignment more concise while preserving the same semantics as creating a named value via `ObjectFactory`:

```kotlin
configurations.resolvable("foo") {
    attributes {
        // Before: 
        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage::class.java, "red"))
        
        // After:

>         attribute(Usage.USAGE_ATTRIBUTE, named(Usage::class.java, "red"))

Author: Yeah, will update the release notes, the Kotlin extension is already merged: https://github.com/gradle/gradle/pull/35246
